### DISCUSSION TITLE
How do I query for where relationship does _not_ exist?

### DISCUSSION THREAD

**Discussion by webnoob** [prisma/prisma#9123] [Q&A] [answered]
I have the following model:

```prisma
model PermissionGroup {
  id          Int      @id @default(autoincrement())
  createdDate DateTime @default(now())
  name        String
  companyPlan Boolean  @default(false)

  updatedDate DateTime?

  company   Company? @relation(fields: [companyId], references: [id])
  companyId Int?

  user   User? @relation(fields: [userId], references: [id])
  userId Int?
}
```
I want to query where companyId matches OR both company and user are null. I thought this should be simple but nothing I am trying is working.

This is my `where` for `findMany`:

```ts
const where = {
      OR: [
        // Company owns this group
        { companyId: tokenUser.companyId },
        {
          companyId: void 0,
          userId: void 0
        }
      ]
    }
```

I've tried variations such as `company: void 0`, `company: { is: void 0 }` but to no avail.

How am I able bring back all permissionGroup where companId = ? or company and user is null?

**Comment by ryands17** [comment, marked as answer]
@webnoob 👋 

This will work:

```ts
  await prisma.permissionGroup.findMany({
    where: {
      OR: [{ companyId: 1 }, { AND: [{ companyId: null }, { userId: null }] }],
    },
  })
```

**Reply by webnoob** [reply]
Hmmm, I avoided using null as normally prisma complains when I try, telling me to use undefined instead 🤔 I did try with the wrapped AND, just using undefined instead.

I'll give it a go. Thanks.

**Reply by ryands17** [reply]
Non-nullable fields do not accept `null`, only nullable fields do.

**Reply by webnoob** [reply]
Yep - that worked. It seems to be OK _without_ the nested AND as well (so just having it like my OP but with null instead of void).

Thanks for your help.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
